src/data/mv_polynomial/basic.lean

```diff
@@ -382,35 +382,36 @@ begin
   rw mul_def,
   have := @finset.sum_sigma (σ →₀ ℕ) R _ _ p.support (λ _, q.support)
     (λ x, if (x.1 + x.2 = n) then coeff x.1 p * coeff x.2 q else 0),
-  convert this.symm using 1; clear this,
+  have aux₁ : (p.support.sigma $ λ _, q.support).filter (λ x, x.1 + x.2 = n) ⊆ _ :=
+    finset.filter_subset _,
+  have aux₂ : (antidiagonal n).support.filter (λ x, x.1 ∈ p.support ∧ x.2 ∈ q.support) ⊆
+             (antidiagonal n).support := finset.filter_subset _,
+  rw [← finset.sum_sdiff aux₁, finset.sum_eq_zero, zero_add] at this, swap,
```

Comment:
```suggestion
  { rw [← finset.sum_sdiff aux₁, finset.sum_eq_zero, zero_add] at this, swap,
```

Can you put braces around the subproof here?  I'm having a hard time to figure out where the have ends.

Reply:
Doesn't that just enclose the rest of the proof in an extra pair of braces and increase the indentation?

Reply:
Right, this is operating in the "main" context. I've added some comments.

Reply:
The `have` cannot be the last tactic in the proof.  Only the part proving the `have` would be indented.

Reply:
Nevermind, I missed the `:=` in the have.

Reply:
The proof of the `have aux₂` is finished by this point though, `finset.filter_subset _` completes it.